src/components/ceremony/ceremony.edgedb.repository.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@nestjs/common';
+import { PublicOf } from '~/common';
+import { RepoFor } from '~/core/edgedb';
+import { CeremonyRepository } from './ceremony.repository';
+import { Ceremony, CreateCeremony } from './dto';
+
+@Injectable()
+export class CeremonyEdgeDBRepository
```

Comment:
Hey @CarsonF - Need help here when you have a moment. I thought that the `static omit = [*defaults*]` section below would fix a lot of the TS issues I am dealing with, but I am still seeing the below error on this CeremonyEdgeDBRepository that I am trying to make sense of: 
```
TS2420: Class CeremonyEdgeDBRepository incorrectly implements interface PublicOf<CeremonyRepository>
Type CeremonyEdgeDBRepository is missing the following properties from type PublicOf<CeremonyRepository>: update, list, readOne, readMany
```

Reply:
The current neo4j repository still has those methods, so we'll need to keep them here until we fully cut over.

Reply:
I guess I misunderstood what this did. I was thinking it omitted them from the customization, not all together. 

So then is there a way to customize the `create` but use the defaults for the rest? It's making me re-create them all to work around TS errors if I customize this one, but I don't feel like I should have to do that.

Reply:
Yes `omit` removes them all together. If you are customizing it you don't need to omit it.
I think there's a bug/annoyance in the TS error. It gives you that, but the real problem is probably the return type of the `create`.
If you do temporarily add all those methods it says are missing and just forward them to the defaults, TS should give you the actual error.

Reply:
It's the `type` that the DTO wants, but EdgeDB schema doesn't have. We can compute it on read, in the `hydrate` method.
```ts
  // Engagement::DedicationCeremony -> Dedication, etc.
  type: castToEnum(ceremony.__type__.name.slice(12, -8), CeremonyType),
```
Just like we do for projects
https://github.com/SeedCompany/cord-api-v3/blob/develop/src/components/project/project.edgedb.repository.ts#L21-L22

Reply:
Added on but still seeing some TS errors. here I now see: 
```
TS2420: Class CeremonyEdgeDBRepository incorrectly implements interface PublicOf<CeremonyRepository>
Type CeremonyEdgeDBRepository is missing the following properties from type PublicOf<CeremonyRepository>: readOne, readMany
```

Reply:
I have literally never struggled with TS errors and how to resolve this much in my life but I can't make sense of these EdgeDB/neo4j mashup TS errors. 😞

Reply:
Ok I totally agree with you that the error it's giving you isn't helpful.
I think it's an intersection between the "`implements` neo4j repo" and the `customize()` signature.
The former is short lived, and I want to rework `customize()` as well. Because I agree it's bad right now.

So I removed the `static omit` you have. And then added
```ts
readOne(id: ID) {
  return this.defaults.readOne(id);
}
readMany(id: readonly ID[]) {
  return this.defaults.readMany(id);
}
```
(which I totally agree you shouldn't need, but this gives the actual type error).
Oh I also changed update to
```ts
async update(changes: UpdateCeremony)
```
to match your neo4j version. (The `session` arg is needed there, but it can be omitted here since it's the last arg and we don't need it here.)

_Finally_ TS tells us the error is

>Types of property `estimatedDate` are incompatible.
>  Type `CalendarDate | null` is not assignable to type `CalendarDate`

This is another long-standing inconsistency we've had.
We declare
https://github.com/SeedCompany/cord-api-v3/blob/1e2dfedf2cc5273df3550df9bc47c1dcaf7349aa/src/components/ceremony/dto/ceremony.dto.ts#L36-L36
which is just a secure wrapped value of `CalendarDate`.

But the EdgeDB schema we've recently declared is `optional` (implicit)
https://github.com/SeedCompany/cord-api-v3/blob/1e2dfedf2cc5273df3550df9bc47c1dcaf7349aa/dbschema/ceremony.esdl#L6-L6
aka nullable.

The neo4j queries don't have strict types nor another schema to compare against, which is why this has lived so long. Now it actually matters as EdgeDB is forcing us to be accurate here.

We need to fix the fields in the DTO to be `SecuredDateNullable` which translates to a secure wrapped value of `CalendarDate | null`.

Hopefully all this makes sense.